### PR TITLE
feat: consolidate to pyproject.toml

### PR DESCRIPTION
**Pull request by WilliamBergamin**
### Type of change (place an x in the [ ] that applies)

- [ ] New sample
- [x] New feature
- [ ] Bug fix
- [ ] Documentation

### Summary

The python community is [slowly recommending](https://packaging.python.org/en/latest/discussions/setup-py-deprecated/#is-pyproject-toml-mandatory) to use the `pyproject.toml` to configure projects

This PR tries to consolidate the project configurations to the `pyproject.toml`

### Requirements (place an x in each [ ] that applies)

- [x] I’ve checked my submission against the Samples Checklist to ensure it complies with all standards
- [x] I have ensured the changes I am contributing align with existing patterns and have tested and linted my code
- [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[flake8]
-exclude = .gitignore,venv
```
Comment (zimeg) [inline setup.cfg:2]: Just checking that this is alright to `exclude` too? I don't find any linter errors with it gone, even with my `.venv` so it seems okay to me

File: .flake8
Comment (zimeg) [inline .flake8:1]: Darn that `flake8` doesn't support `pyproject.toml` yet 😕

**Review by zimeg — APPROVED** [review]
LGTM! 🐍 The same configurations in fewer files is always a wonderful thing

File: setup.cfg (hunk 1 above)
Comment (WilliamBergamin) [reply setup.cfg:2]: Actually Flake8 does not exclude the virtual env folders, the dev must exclude them manually based on what they named the folders `.env` `.venv` `venv` etc.. 

This is why the readme informs the users to use `flake8 *.py && flake8 listeners/` in order to lint the project

**Pull request merged**